dimod/core/sampler.py

```diff
@@ -238,3 +239,23 @@ def sample_qubo(self, Q, **parameters):
         """
         bqm = BinaryQuadraticModel.from_qubo(Q)
         return self.sample(bqm, **parameters)
+
+    def validate_kwargs(self, **kwargs):
+        """Check that all `kwargs` are accepted by the sampler. If a
+        keyword is unknown, a warning is raised and the argument is removed.
+
+        Args:
+            **kwargs:
+                Keyword arguments to be validated.
+
+        Returns:
+            dict: Updated `kwargs`
+
+        """
+        for kw in kwargs.copy():
```

Comment:
Better to do `kw in list(kwargs)` rather than copying the full thing. Shouldn't make much difference because it's a shallow copy, but might as well only make copies of the things we care about (the keys).

Reply:
Or even
```
for kw in [kw for kw in kwargs if kw not in self.parameters]:
    ....
```